### PR TITLE
refactor(typescript): set autocmd with AstroLSP

### PR DESCRIPTION
**Pull request by mehalter**
## 📑 Description

This uses the new notation in AstroLSP for setting up autocommands when attaching a language server. This makes it easy for users to opt out of the "ESLintFixAll" autocommand with something like this in their user plugins:

```lua
return {
  "AstroNvim/astrolsp",
  opts = {
    autocmds = {
      eslint_fix_on_save = false
    }
  }
}
```

### CODE REVIEW

**Comment by mehalter** [comment]
I also checked the rest of the AstroCommunity specs and this is the only autocommand that is applicable for this new notation

**Comment by Uzaaft** [comment]
> I also checked the rest of the AstroCommunity specs and this is the only autocommand that is applicable for this new notation

Damn this is neato!

File: lua/astrocommunity/pack/typescript/init.lua [hunk 1]
```diff
@@ -105,6 +88,16 @@ return {
       "AstroNvim/astrolsp",
       ---@diagnostic disable: missing-fields
       opts = {
+        autocmds = {
+          eslint_fix_on_save = {
+            cond = function(client) return client.name == "eslint" and vim.fn.exists ":EslintFixAll" > 0 end,
```
Comment (mehalter) [inline lua/astrocommunity/pack/typescript/init.lua:93]:
```suggestion
            cond = function() return vim.fn.exists ":EslintFixAll" > 0 end,
```

Maybe we should just simplify it to this @Uzaaft what do you think?

**Comment by Uzaaft** [comment]
@mehalter  I added some basic docs into this PR

File: lua/astrocommunity/pack/typescript/init.lua (hunk 1 above)
Comment (Uzaaft) [reply lua/astrocommunity/pack/typescript/init.lua:93]: I have zero opinions about this 🙊

Comment (mehalter) [reply lua/astrocommunity/pack/typescript/init.lua:93]: Ok, let's just leave it as is

**Review by Uzaaft — APPROVED** [review]

**Comment by Uzaaft** [comment]
LGTM 🚀

**Pull request merged**